tools/graph-convert/tests/graph-properties-convert-test.cpp

```
@@ -1100,49 +1099,49 @@ galois::GraphComponents GenerateAndConvertBson(size_t chunk_size) {
 
   bson_t* george = BCON_NEW("_id", BCON_OID(&george_oid), "name",
                             BCON_UTF8("George"), "born", BCON_DOUBLE(1985));
-  galois::HandleNodeDocumentMongoDB(&builder, &properties, george, "person");
+  galois::HandleNodeDocumentMongoDB(&builder, george, "person");
   bson_t* frank = BCON_NEW("_id", BCON_OID(&frank_oid), "name",
                            BCON_UTF8("Frank"), "born", BCON_DOUBLE(1989));
-  galois::HandleNodeDocumentMongoDB(&builder, &properties, frank, "person");
+  galois::HandleNodeDocumentMongoDB(&builder, frank, "person");
 
   bson_t* friend_doc =
       BCON_NEW("_id", BCON_OID(&friend_oid), "friend1", BCON_OID(&george_oid),
                "friend2", BCON_OID(&frank_oid), "met", BCON_DOUBLE(2000));
-  galois::HandleEdgeDocumentMongoDB(&builder, &properties, friend_doc,
-                                    "friend");
+  galois::HandleEdgeDocumentMongoDB(&builder, friend_doc, "friend");
 
-  return galois::BuildGraphComponents(builder, properties);
+  return builder.Finish();
+#endif
 }
 
 void VerifyMongodbSet(const galois::GraphComponents& graph) {
 #ifndef GALOIS_MONGOC_FOUND
   (void)graph;
   return;
-#endif
-  GALOIS_ASSERT(graph.node_properties->num_columns() == 2);
-  GALOIS_ASSERT(graph.node_labels->num_columns() == 1);
-  GALOIS_ASSERT(graph.edge_properties->num_columns() == 1);
-  GALOIS_ASSERT(graph.edge_types->num_columns() == 1);
+#else
```

some preference for simplifying the conditional preprocessing by pulling out mongodb stuff to a separate file. In any event, adding something like
```cpp
int main(...) {
...

switch (file_type) {
...
#ifdef GALOIS_MONGOC_FOUND
  case galois::SourceDatabase::kMongodb:
    ...
#endif
  default:
    GALOIS_LOG_FATAL(...);
}
...

}
```

will ensure that we fail the test if we try to run a mongodb test when we are built without mongodb rather than silently succeeding.

---

the point of adding it so that mongodb did not build was so that we could pass python tests...

---

You could just add something to tools/.../tests/CMakeLists.txt like:
```cmake
if(mongoc-1.0_FOUND)
  add_test(NAME convert-properties-mongodb ...)
else()
  message(STATUS "Skipping mongodb tests")
endif()
```

More work, I know, but in the off chance someone has a `graph-properties-convert-test` program and runs it with `--mongodb` manually without mongodb support, I think would be better to report an error than to silently return success.

---

so I just add that message? that would not be much work?